### PR TITLE
[Ide] Implement cancellation via status bar.

### PR DESCRIPTION
**Pull request by Therzok**
This adds a basis for searchbar cancellation via the status bar. The
CancellationTokenSource is passed to the statusbar implementation to be
used to cancel the current operation.

The implementation is only done for Mac.

Bug 41335 - StatusBar cancellation

cc @slluis on API changes.
cc @iainx on Mac changes.
cc @vancura on UI changes.

### CODE REVIEW

**Comment by Therzok** [comment]
There currently is a problem where we have the following situation:
* A monitor does not set its own cancellationtokensource.
* One is created by default for it [here](https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Core/MonoDevelop.Core/ProgressMonitor.cs#L542-L543) - one example is loading a solution
* The cancel button appears but does nothing since the operation doesn't handle cancellationtokensource cancellation.

Maybe we should default it to not construct a source by default and add nullchecks on where we work on it?

**Comment by iainx** [comment]
Mac bits look good to me 👍

**Comment by vancura** [comment]
Design bit looks good to me 👍

**Comment by slluis** [comment]
The button should have some kind of visual feedback when hovering.

**Comment by slluis** [comment]
@Therzok I don't understand the issue you are explaining. Can you elaborate?

**Comment by Therzok** [comment]
@slluis We decided it should be opt-in, not on by default. The issue is that not all operations have a cancellation handler.

**Comment by Therzok** [comment]
TODO: hover effect.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs [hunk 1]
```diff
@@ -124,9 +124,9 @@ protected override OperationConsole OnCreateConsole (CreateConsoleOptions option
 		/******************************/
 		
 		
-		public ProgressMonitor GetStatusProgressMonitor (string title, IconId icon, bool showErrorDialogs, bool showTaskTitle = true, bool lockGui = false, Pad statusSourcePad = null)
+		public ProgressMonitor GetStatusProgressMonitor (string title, IconId icon, bool showErrorDialogs, bool showTaskTitle = true, bool lockGui = false, Pad statusSourcePad = null, bool showCancelButton = false)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs]: You can't just add an optional parameter. This is an API break.

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs]: Will fix.

Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs]: For now you can just add an overload that requires all parameters.

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs]: That was my plan, thanks.

**Comment by Therzok** [comment]
Updated.

**Pull request merged**